packages/block-editor/src/components/use-block-commands/index.js

```diff
@@ -283,6 +292,48 @@ const getQuickActionsCommands = () =>
 			} );
 		}
 
+		if ( canToggleBlockVisibility ) {
+			const hasHiddenBlock = blocks.some(
+				( block ) =>
+					block.attributes.metadata?.blockVisibility === false
+			);
+			const blockVisibilityLabel = hasHiddenBlock
+				? _n(
+						'Show selected block',
+						'Show selected blocks',
+						blocks.length
+				  )
+				: _n(
+						'Hide selected block',
+						'Hide selected blocks',
+						blocks.length
+				  );
+
+			commands.push( {
+				name: 'core/toggle-block-visibility',
+				label: blockVisibilityLabel,
+				callback: () => {
+					const attributesByClientId = Object.fromEntries(
```

Comment:
A lot of this is copied over from:

https://github.com/WordPress/gutenberg/blob/update/block-visibility-command/packages/block-editor/src/components/block-visibility/menu-item.js#L18

https://github.com/WordPress/gutenberg/blob/update/block-visibility-command/packages/block-editor/src/components/block-visibility/toolbar.js#L59

Is there scope for centralization, e.g., a private action?